.github/guides/STANDARDS.md

```diff
@@ -310,17 +310,139 @@ https://file.house/zy7H.png
+		. = "were"
+```
+
+Because of these problems, it is encouraged to prefer standard, explicit return statements. The above code would be best written as:
+
+```dm
+/client/p_were(gender)
+	if (gender == PLURAL || gender == NEUTER)
+		return "were"
+	else
+		return "was"
+```
+
+#### Exception: `. = ..()`
+
+As hinted at before, `. = ..()` is *extremely* common. This will call the parent function, and preserve its return type. Code like this:
+
+```dm
+/obj/item/spoon/attack()
+	. = ..()
+	visible_message("Whack!")
+```
+
+...is completely accepted, and in fact, usually *prefered* over:
+
+```dm
+/obj/item/spoon/attack()
+	var/output = ..()
+	visible_message("Whack!")
+	return output
+```
+
+#### Exception: Runtime resilience
+
+One unique property of DM is the ability for procs to error, but for code to continue. For instance, the following:
+
+```dm
+/proc/uh_oh()
+	CRASH("oh no!")
+
+/proc/main()
+	to_chat(world, "1")
+	uh_oh()
+	to_chat(world, "2")
 ```
-However, DM also has a dot variable, accessed just as `.` on its own, defaulting to a value of null. Now, what's special about the dot operator is that it is automatically returned (as in the `return` statement) at the end of a proc, provided the proc does not already manually return (`return count` for example.) Why is this special?
 
-With `.` being everpresent in every proc, can we use it as a temporary variable? Of course we can! However, the `.` operator cannot replace a typecasted variable - it can hold data any other var in DM can, it just can't be accessed as one, although the `.` operator is compatible with a few operators that look weird but work perfectly fine, such as: `.++` for incrementing `.'s` value, or `.[1]` for accessing the first element of `.`, provided that it's a list.
+...would print both 1 *and* 2, which may be unexpected if you come from other languages.
+
+This is where `.` provides a new useful behavior--**a proc that runtimes will return `.`**.
+
+Meaning:
+
+```dm
+/proc/uh_oh()
+	. = "woah!"
+	CRASH("oh no!")
+
+/proc/main()
+	to_chat(world, uh_oh())
+```
+
+...will print `woah!`. 
+
+For this reason, it is acceptable for `.` to be used in places where consumers can reasonably continue in the event of a runtime.
+
+If you are using `.` in this case (or for another case that might be acceptable, other than most uses of `. = ..()`), it is still prefered that you explicitly `return .` in order to prevent both editor issues and readability/error-prone issues.
+
+```dm
+/proc/uh_oh()
+	. = "woah!"
+
+	if (do_something())
+		call_code()
+		if (!working_fine())
+			return . // Instead of `return`, we explicitly `return .`
+
+	if (some_fail_state())
+		CRASH("youch!")
+
+	return . // `return .` is used at the end, to signify it has been used
+```
+
+```dm
+/obj/item/spoon/attack()
+	. = ..()
+	if (!.)
+		return . // `. = ..()` does not require us to `return .` at the end, but in this case, we should make it clear what is being returned here
```

Comment:
`return .` is cringe in any circumstance tbh

Reply:
Treating falsy values as interchangable is how you get weird bugs where `isnull` fails and `?.` runtimes. **Explicitness is better in most cases.** I enforce making sure procs always return the correct values depending on their name--`0` when it's a number (such as `get_number_of_xxx`), `FALSE` when it's a boolean, and `null` when it's nothing.

Reply:
in this case I think it's important that you should be explicitly returning FALSE, or looking at what attack() is doing, because I really don't like how `return .` looks

Reply:
I don't like `return .` it looks dumb.

Reply:
Echoing the other's thoughts, I know there's a valid argument about being explicit vs implicit but I cannot will myself to use `return .`

Reply:
I don't think the implicit return is always bad, it's more that code that relies on it heavily is likely just being needlessly obtuse.

That doesn't mean that . = ..() if(!.) return is bad, but . = TRUE - LONG CHAIN OF LOGIC return is. At that point you're starting to obscure the intentional return value significantly, rather then simply using . as an implicit return. I think that's the issue here.

There are actually even some instances where . = TRUE would be ok, ui_act's with behavior after the switch come to mind (not sure what they even put there, but I've seen the pattern before)

Reply:
TL;DR If you make me write return . I will jump off a building

Reply:
Either way is fine I'm not bothered

Reply:
```dm
if (!.)
    return
```

...gains nothing over...

```dm
if (!.)
    return .
```

Ultimately, I would prefer just directly `return FALSE` or `return null` in most circumstances, however.

Stuff like `ui_act` setting `. = TRUE` at the beginning and forgetting about it is still fairly problematic. People don't actually know what `ui_act`'s return *does*. Sure, nobody is creating a new consumer for ui_act anytime soon, but explicit returns make it extremely clear what is happening. I always did `return TRUE` and whatnot explicitly on ui_act. In fact, if what you really want is just cleanliness, you can just shove `return TRUE` at the end of ui_act, which is what I'm doing anyway.

I can understand the hesitancy in *this example*, but `return .` is the only sane way to promote both runtime resilience without having all the baggage `.` normally carries (impossible to directly go to, makes it unclear what a proc is returning)

Reply:
![image](https://user-images.githubusercontent.com/35135081/154334803-7998e333-1d56-4392-ba60-05b81357250e.png)

Reply:
Correction: `return .` at the end is still encouraged by this doc, because it makes it clear that the proc actually returns something **which is a critical component to a proc's signature**.

Reply:
I removed this specific example because this one already conveys the point better, but `return .` is absolutely important.

![image](https://user-images.githubusercontent.com/35135081/154338325-1bf8fed1-044a-4beb-9c70-5b6534543dac.png)